### PR TITLE
Change locale of mdspell to en-US

### LINKED ISSUE #6736: Convert to using mdspell with en-US
[CONTRIBUTING.md](https://github.com/PowerShell/PowerShell/blob/fc627a1dba69ad839fcb7fe76b9aa4733e2b371f/.github/CONTRIBUTING.md#spellchecking-documentation) specifies that we use `mdspell` to spellcheck our documentation.

But [`mdspell`'s default locale is en-GB](https://www.npmjs.com/package/markdown-spellcheck#british-english----en-gb-), meaning there is a [number](https://github.com/PowerShell/PowerShell/blob/fc627a1dba69ad839fcb7fe76b9aa4733e2b371f/.spelling#L31) [of](https://github.com/PowerShell/PowerShell/blob/fc627a1dba69ad839fcb7fe76b9aa4733e2b371f/.spelling#L279) [false](https://github.com/PowerShell/PowerShell/blob/fc627a1dba69ad839fcb7fe76b9aa4733e2b371f/.spelling#L512) [positives](https://github.com/PowerShell/PowerShell/blob/fc627a1dba69ad839fcb7fe76b9aa4733e2b371f/.spelling#L47) in the spellchecker. It's a small issue, but easy enough to fix too.

### PR DESCRIPTION
**Pull request by adityapatwardhan**
Fix #6736

## PR Summary

Update the locale to use en-us for mdspell checker. 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by anmenaga** [comment]
What is bad about the old code? How did it manifest itself?

**Review by anmenaga — APPROVED** [review]

**Comment by rjmholt** [comment]
Looks like Travis is failing because of the spellcheck. Maybe worth regenerating the spelling exceptions file for this PR?

File: .spelling [hunk 1]
```diff
@@ -3,1115 +3,790 @@
...
-cataloged
-CentOS
-childitem
-chucklu
-cimsession
-cimsupport
-cmake
-cmd
+powershell
+changelog
```
Comment (TravisEz13) [inline .spelling]: Can you resort and update the comments base on your change?

Comment (adityapatwardhan) [reply .spelling]: Sorted the file as requested.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**